BUILDER_DEV.md

```diff
@@ -1,225 +1,113 @@
-
+## Pre-Requisites
 
-### Builder Setup
-* Update your system
-```
-$ sudo apt-get update
-```
-
-* NOTE - if you are running Ubuntu 17.10 Server, the above command may fail with "E: The repository 'cdrom://Ubuntu-Server 17.10 _Artful Aardvark_ - Release amd64 (20171017.1) artful Release' does not have a Release file."  To fix this, run:
-
-```
-$ sudo sed -i.bak '/deb cdrom/s/^/#/g' /etc/apt/sources.list
-```
+### Ports required
+1. 9636 - Intra-supervisor communication
+1. 80 - Web app
+1. 9631 - supervisor api port
+1. 5432 - posgres port
 
-* Install git
-```
-$ sudo apt-get install git
-```
+### Checkout
+1. `git clone https://github.com/habitat-sh/habitat.git`
 
-* Clone the [Habitat repo](https://github.com/habitat-sh/habitat)
-```
-$ git clone https://github.com/habitat-sh/habitat.git
-```
+If you are using Linux environment you can run `support/linux/provision.sh` to configure your host
+If you are on a Mac, you will need brew, direnv, habitat, and Docker for Mac
 
-* Copy habitat-builder-dev.2017-10-02.private-key.pem from 1Password into path/to/habitat/repo/.secrets/builder-github-app.pem (NOTE - this is required as of 12/11/17 - we are working on a solution to make access to the Habitat 1Password vault not required for dev environment setup).
-* Set up some environmental variables (I put these in my .bashrc)
+### GitHub OAuth Application
+`APP_HOSTNAME` mentioned below, will typically be `localhost`
 
-```
-export HAB_AUTH_TOKEN="<your github token>"
-export HAB_BLDR_URL="http://localhost:9636"
-export HAB_ORIGIN="core"
-```
+1. [Setup a GitHub application](https://github.com/settings/apps/new) for your GitHub organization.
+1. Set the value of `Homepage URL` to `http://${APP_HOSTNAME}`
+1. Set the value of `User authorization callback URL` to `http://${APP_HOSTNAME}/` (The trailing `/` is *important*)
+1. Set the value of `Webhook URL` to `http://${APP_HOSTNAME}/`
+1. Set Repository metadata, Repository administration, Repository content and Organization members to read only (this is only used for your org so it's safe)
+1. Save and download the private key. It will result in a file like `app-name.date.private-key.pem`
+1. Copy the private key to `${HABITAT_SRC_ROOT}/.secrets/builder-github-app.pem`
+1. Record the the client-id, client-secret, app_id and webhook-secret. These will be used for the `GITHUB_CLIENT_ID`, `GITHUB_CLIENT_SECRET` and `GITHUP_APP_ID` build variables (seen below).
 
-* symlink /src to path/to/your/habitat/repo
+### Create app env file
 
+1. Create `${HABITAT_SRC_ROOT}/.secrets/habitat-env`
+1. Add
 ```
-$ sudo ln -s path/to/your/habitat/repo /src
+  APP_HOSTNAME=localhost
+  GITHUB_API_URL=https://api.github.com
+  GITHUB_WEB_URL=https://github.com
+  GITHUB_CLIENT_ID=${GITHUB_CLIENT_ID}
+  GITHUB_CLIENT_SECRET=${GITHUB_CLIENT_SECRET}
+  GITHUB_APP_ID=${GITHUP_APP_ID}
+  GITHUB_APP_NAME="my-dev-app"
```

Comment:
`my-dev-app` has to correspond to the name that was entered for `GitHub App name`. So we could suggest people hardcode something, or just say that it has to be whatever they named it processed through `tr "[:upper:]" "[:lower:]" | tr " " "-"`